### DISCUSSION TITLE
may i train a RAG model based on my local database without  a api_key?

### DISCUSSION THREAD

**Discussion by davidhandsome86** [vanna-ai/vanna#267] [Q&A] [answered]
first, thanks a lot for this  awesome package, when i use vanna to train my local database like this :

df_information_schema = vn.run_sql("SELECT ***************")
plan = vn.get_training_plan_generic(df_information_schema)
plan

but it failed ,like this:
![image](https://github.com/vanna-ai/vanna/assets/150113317/a4bdece2-67d7-449f-98cc-00ec899ada5d)

my question is: does vanna can train and ask without a api_key ?  i want to train and save a RAG model offline.
please let me know how to do that .

thanks a lot

**Comment by zainhoda** [comment, marked as answer]
Hi -- it's not necessary to use a Vanna API key if you choose your own LLM and ChromaDB, which runs locally:

Here's an example in the documentation:
https://vanna.ai/docs/postgres-ollama-chromadb/

You can just click to change the configuration
